contracts/governance/Vesting/VestingRegistryLogic.sol

```diff
@@ -182,6 +183,15 @@ contract VestingRegistryLogic is VestingRegistryStorage {
                 uint256(VestingType.Vesting),
                 _vestingCreationType
             );
+
+        vestingDetails[vesting] = Vesting(
+            uint256(VestingType.Vesting),
+            _vestingCreationType,
+            vesting
```

Comment:
it is redundant to have vesting address as a value in the mapping when it is already the key. 

the Vesting struct is catastrophic from a gas perspective. We sadly can't change it anymore, but when we introduce a new struct for the details, then we make sure that the content will fit into a single storage slot. it is safe to assume that we will never have `2^256` vesting types or vesting creation types.

Reply:
and maybe rename from vestingDetails to vestingTypes, else i'd expect to get all of the details like duration, cliff, token owner etc

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/21dd284dc55c0b0355bb165ce2e24dbf154e32e6)

Introduced new struct variable